### PR TITLE
pspp: Update to v2.0.1

### PR DESCRIPTION
**Pull request by malfisya**
**Summary**

Full changelog can be read [here](http://git.savannah.gnu.org/cgit/pspp.git/plain/NEWS?h=v2.0.1)

Resolves getsolus/packages#4175

**Test Plan**

Launch `pspp`, `psppire`

**Checklist**

- [x] Package was built and tested against unstable

### CODE REVIEW

File: packages/p/pspp/pspec_x86_64.xml [hunk 1]
```diff
@@ -21,16 +22,13 @@
...
-            <Path fileType="executable">/usr/bin/pspp-dump-sav</Path>
             <Path fileType="executable">/usr/bin/pspp-output</Path>
             <Path fileType="executable">/usr/bin/psppire</Path>
-            <Path fileType="library">/usr/lib64/pspp/libpspp-1.4.1.so</Path>
-            <Path fileType="library">/usr/lib64/pspp/libpspp-core-1.4.1.so</Path>
+            <Path fileType="library">/usr/lib64/pspp/libpspp-2.0.1.so</Path>
+            <Path fileType="library">/usr/lib64/pspp/libpspp-core-2.0.1.so</Path>
             <Path fileType="library">/usr/lib64/pspp/libpspp-core.so</Path>
             <Path fileType="library">/usr/lib64/pspp/libpspp.so</Path>
-            <Path fileType="data">/usr/share/applications/org.fsf.pspp.desktop</Path>
```
Comment (malfisya) [inline packages/p/pspp/pspec_x86_64.xml:31]: Not Ideal

File: packages/p/pspp/pspec_x86_64.xml [hunk 2]
```diff
@@ -45,31 +43,26 @@
...
             <Path fileType="data">/usr/share/icons/hicolor/96x96/mimetypes</Path>
-            <Path fileType="data">/usr/share/icons/hicolor/scalable/apps/pspp.svg</Path>
+            <Path fileType="data">/usr/share/icons/hicolor/scalable/apps/org.gnu.pspp.svg</Path>
             <Path fileType="data">/usr/share/icons/hicolor/scalable/mimetypes</Path>
-            <Path fileType="info">/usr/share/info/pspp-dev.info</Path>
-            <Path fileType="info">/usr/share/info/pspp-dev.info-1</Path>
-            <Path fileType="info">/usr/share/info/pspp-dev.info-2</Path>
-            <Path fileType="info">/usr/share/info/pspp.info</Path>
-            <Path fileType="info">/usr/share/info/pspp.info-1</Path>
-            <Path fileType="info">/usr/share/info/pspp.info-2</Path>
```
Comment (malfisya) [inline packages/p/pspp/pspec_x86_64.xml:72]: Not Ideal

File: packages/p/pspp/pspec_x86_64.xml [hunk 3]
```diff
@@ -85,16 +78,14 @@
...
+            <Path fileType="localedata">/usr/share/locale/ta/LC_MESSAGES/pspp.mo</Path>
             <Path fileType="localedata">/usr/share/locale/tr/LC_MESSAGES/pspp.mo</Path>
             <Path fileType="localedata">/usr/share/locale/uk/LC_MESSAGES/pspp.mo</Path>
             <Path fileType="localedata">/usr/share/locale/zh_CN/LC_MESSAGES/pspp.mo</Path>
             <Path fileType="man">/usr/share/man/man1/pspp-convert.1</Path>
             <Path fileType="man">/usr/share/man/man1/pspp-dump-sav.1</Path>
             <Path fileType="man">/usr/share/man/man1/pspp-output.1</Path>
-            <Path fileType="man">/usr/share/man/man1/pspp.1</Path>
-            <Path fileType="man">/usr/share/man/man1/psppire.1</Path>
-            <Path fileType="data">/usr/share/metainfo/org.fsf.pspp.metainfo.xml</Path>
```
Comment (malfisya) [inline packages/p/pspp/pspec_x86_64.xml:96]: Bad

**Review by malfisya — COMMENTED** [review]
Disabling doc cause side effect of losing desktop file, metainfo, manpages, and few other things. Might look into adding it manually.

**Comment by Staudey** [comment]
Using just these parts of the doc/automake.mk brings the desktop file and man pages back at least:

```
dist_man_MANS += doc/pspp.1 \
                 doc/psppire.1

desktopdir = $(datadir)/applications

doc/org.gnu.pspp.metainfo.xml: doc/org.gnu.pspp.metainfo.xml.in $(POFILES)
	$(AM_V_GEN)$(MSGFMT) --xml --template $< -o $@ -d $(top_srcdir)/po || \
	  $(MSGFMT) -L appdata --xml --template $< -o $@ -d $(top_srcdir)/po

doc/org.gnu.pspp.desktop: doc/org.gnu.pspp.desktop.in $(POFILES)
	$(AM_V_GEN)$(MSGFMT) --desktop --template $< -o $@ -d $(top_srcdir)/po

CLEANFILES+=doc/org.gnu.pspp.desktop \
            doc/org.gnu.pspp.metainfo.xml

desktop_DATA = doc/org.gnu.pspp.desktop

appdatadir = $(datadir)/metainfo
dist_appdata_DATA = doc/org.gnu.pspp.metainfo.xml

EXTRA_DIST += doc/org.gnu.pspp.metainfo.xml.in \
	doc/org.gnu.pspp.desktop.in
```

(In my testing I simply added it to the package files and then copied it over doc/automake.mk)

File: packages/p/pspp/package.yml [hunk 4]
```diff
@@ -1,19 +1,27 @@
...
     - pkgconfig(gsl)
-    - pkgconfig(gtksourceview-3.0)
+    - pkgconfig(gtksourceview-4)
+    - pkgconfig(libpng)
     - pkgconfig(spread-sheet-widget)
+    - gettext-devel
 setup      : |
-    %configure --disable-static
+    # See http://savannah.gnu.org/bugs/?65892, use Alpine patch to disable doc
+    #patch -p1 -i $pkgfiles/disable-doc.patch
```
Comment (malfisya) [inline packages/p/pspp/package.yml:20]: End up not using it. Should I remove it? It still provide contexts to all patches.

Comment (Staudey) [reply packages/p/pspp/package.yml:20]: Personally I'd leave the reference to the upstream bug report above the new patch.

Something like `# Workaround for http://savannah.gnu.org/bugs/?65892`

I don't see a reason to keep the "old" disable-doc.patch file around though

File: packages/p/pspp/monitoring.yml [hunk 5]
```diff
@@ -0,0 +1,8 @@
+releases:
+  id: 3722
+  rss: http://savannah.gnu.org/news/atom.php?group=pspp
+# No known CPE, checked 2024-11-11
```
Comment (davidjharder) [inline packages/p/pspp/monitoring.yml:4]: It appears we know the CPE :)

**Review by davidjharder — APPROVED** [review]
Looks good! Great job on a tough homepage straggler.

**Pull request merged**